sdk/Sdk.Analyzers/RegistrationExpectedInASPNetIntegration.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class RegistrationExpectedInASPNetIntegration : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DiagnosticDescriptors.CorrectRegistrationExpectedInAspNetIntegration);
+
+        private const string ExpectedRegistrationMethod = "ConfigureFunctionsWebApplication";
+        private const string AspNetExtensionAssemblyName = "Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore";
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            context.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Namespace);
+        }
+
+        private static void AnalyzeMethod(SymbolAnalysisContext context)
+        {
+            bool isAspNetAssembly = context.Compilation.ReferencedAssemblyNames.Where(assembly => assembly.Name.Equals(AspNetExtensionAssemblyName)).Any();
+            SyntaxReference syntaxReference = context.Symbol.DeclaringSyntaxReferences.FirstOrDefault();
+            string code = syntaxReference.SyntaxTree.GetText().ToString();
+
```

Comment:
I think we do not need to scan the entire source code from the app. So basically, exit if the current method we are analyzing is not the `main` method of that assembly.

Reply:
I was scanning completely because it's not necessary that there would be a `Main` method like the code sample [here]( https://github.com/Azure/azure-functions-dotnet-worker/blob/main/samples/AspNetIntegration/Program.cs). Is there any way to detect this implicit `Main`?

Reply:
Updated code to check only Main method now